diff --git a/Python/pylifecycle.c b/Python/pylifecycle.c
index 1701a1cd..802f2b09 100644
--- a/Python/pylifecycle.c
+++ b/Python/pylifecycle.c
@@ -1569,6 +1569,7 @@ finalize_remove_modules(PyObject *modules, int verbose)
                 PyObject *value = PyObject_GetItem(modules, key);
                 if (value == NULL) {
                     PyErr_FormatUnraisable("Exception ignored on removing modules");
+                    Py_DECREF(key);
                     continue;
                 }
                 CLEAR_MODULE(key, value);
